system/jlib/CMakeLists.txt

```diff
@@ -30,6 +30,8 @@ CHECK_LIBRARY_EXISTS(dl dlopen "" HAVE_LIBDL)
 CHECK_LIBRARY_EXISTS(crypt crypt "" HAVE_LIBCRYPT) 
 endif ()
 
+SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wno-switch -Wno-unused-parameter -Werror")
```

Comment:
NOTE: For the first few files I was also trying to suppress unused-parameter errors. In two minds about that one.